### PR TITLE
CATL-1927: Fix modal fullscreen height issue

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
jQuery UI modals do not take the full height of the screen, when making it fullscreen. But now we want the popups to take full height also. So this PR implements that.

## Before
2020-11-03 at 3 10 PM

## After
2020-11-03 at 3 09 PM

## Technical Details
1. Added a `js/jquery-ui-popup.js` file, which
   * Adds a `fullscreen` class when expand button is pressed.
   * Also assigns the height of the `ui-dialog-content` dynamically.
```javascript
        var menuHeight = $('#civicrm-menu').outerHeight();
	var titleBarHeight = $el.parent().find('.ui-dialog-titlebar').outerHeight();
	var buttonContainerHeight = $el.parent().find('.ui-dialog-buttonpane').outerHeight();
	var windowHeight = $(window).outerHeight();
	
	// this is the height we are setting
	var contentHeight = windowHeight - (titleBarHeight + menuHeight + buttonContainerHeight);
```
2. In `scss/jquery/overrides/_ui-dialog.scss`, added styles specific to `fullscreen` class, to apply full height.

## Comment
Backstop JS tests were not run, because currently we dont have any scenarios written for expanding modals.

### CODE REVIEW

**Comment by swastikpareek** [comment]
@deb1990 : Could we please mention in the technical details why we need to add `js/jquery-ui-popup.js`? Do you think we should add details how are we doing this `Also assigning the height of the ui-dialog-content dynamically` ?

Also, I think the overview needs to be refactored to follow[ PR guidelines](https://compucorp.atlassian.net/wiki/spaces/SD/pages/1520304185/Pull+Requests+guidelines)

File: scss/jquery/overrides/_ui-dialog.scss [hunk 1]
```diff
@@ -3,3 +3,23 @@
...
+#{$outside-namespace-marker} {
+  .crm-container.ui-dialog {
+    &.fullscreen {
+      bottom: 0;
+      max-height: none !important;
+
+      .ui-dialog-buttonpane {
+        bottom: 0;
+        left: 0;
+        position: absolute !important;
```
Comment (swastikpareek) [inline scss/jquery/overrides/_ui-dialog.scss:16]: Please add a reason to why we are adding `!important` tag. Can't it be solved by w/o using `!important` tag?

File: scss/jquery/overrides/_ui-dialog.scss [hunk 2]
```diff
@@ -3,3 +3,23 @@
...
+        right: 0;
+      }
+
+      .ui-dialog-content {
+        max-height: none !important;
```
Comment (swastikpareek) [inline scss/jquery/overrides/_ui-dialog.scss:21]: Same here.

File: scss/jquery/overrides/_ui-dialog.scss [hunk 3]
```diff
@@ -3,3 +3,23 @@
 .ui-dialog {
   @include ui-dialog;
 }
+
+#{$outside-namespace-marker} {
+  .crm-container.ui-dialog {
+    &.fullscreen {
+      bottom: 0;
+      max-height: none !important;
```
Comment (swastikpareek) [inline scss/jquery/overrides/_ui-dialog.scss:11]: Same here.

File: scss/jquery/overrides/_ui-dialog.scss (hunk 1 above)
Comment (deb1990) [reply scss/jquery/overrides/_ui-dialog.scss:16]: This is actually not required, thanks for flagging. Removed.

File: scss/jquery/overrides/_ui-dialog.scss (hunk 3 above)
Comment (deb1990) [reply scss/jquery/overrides/_ui-dialog.scss:11]: This is actually not required, thanks for flagging. Removed.

File: scss/jquery/overrides/_ui-dialog.scss (hunk 2 above)
Comment (deb1990) [reply scss/jquery/overrides/_ui-dialog.scss:21]: Added comment on code.

File: js/jquery-ui-popup.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * This script is responsible for making the jQuery UI popups take full height
+ * of the screen in fullscreen mode.
+ */
+
+(function ($) {
+  $(document).on('dialogopen', function(e) {
+    var $el = $(e.target);
+
+    if ($el.parent().hasClass('crm-container') && $el.dialog('option', 'resizable')) {
+      $('.crm-dialog-titlebar-resize', $el.parent()).click(function(e) {
```
Comment (swastikpareek) [inline js/jquery-ui-popup.js:11]: I see multiple usages of `$el.parent()` in the code. Should we create a variable for this to stop unnecessary DOM queries?

**Review by swastikpareek — APPROVED** [review]

**Pull request merged**